src/framework/_flex.scss

```diff
@@ -0,0 +1,361 @@
+    justify-content: space-between;
+  }
+
+  .justify-evenly-self {
+    justify-content: space-evenly;
+  }
+
+  @include for-each-breakpoints using ($breakpoint) {
+    .justify-start--#{$breakpoint} {
+      justify-content: flex-start;
+    }
+
+    .justify-center--#{$breakpoint} {
+      justify-content: center;
+    }
+
+    .justify-end--#{$breakpoint} {
+      justify-content: flex-end;
+    }
+
+    .justify-around--#{$breakpoint} {
+      justify-content: space-around;
+    }
+
+    .justify-between--#{$breakpoint} {
+      justify-content: space-between;
+    }
+
+    .justify-evenly--#{$breakpoint} {
+      justify-content: space-evenly;
+    }
+
+    .justify-start-self--#{$breakpoint} {
+      justify-content: flex-start;
+    }
+
+    .justify-center-self--#{$breakpoint} {
+      justify-content: center;
+    }
+
+    .justify-end-self--#{$breakpoint} {
+      justify-content: flex-end;
+    }
+
+    .justify-around-self--#{$breakpoint} {
+      justify-content: space-around;
+    }
+
+    .justify-between-self--#{$breakpoint} {
+      justify-content: space-between;
+    }
+
+    .justify-evenly-self--#{$breakpoint} {
+      justify-content: space-evenly;
+    }
+  }
+}
+
+/*============================================================================*\
+   Direction
+\*============================================================================*/
+@if $has-classes {
+  .flex-row {
+    flex-direction: row;
+  }
+
+  .flex-row-reverse {
+    flex-direction: row-reverse;
+  }
+
+  .flex-column {
+    flex-direction: column;
+  }
+
+  .flex-column-reverse {
+    flex-direction: column-reverse;
+  }
+
+  @include for-each-breakpoints using ($breakpoint) {
+    .flex-row--#{$breakpoint} {
+      flex-direction: row;
+    }
+
+    .flex-row-reverse--#{$breakpoint} {
+      flex-direction: row-reverse;
+    }
+
+    .flex-column--#{$breakpoint} {
+      flex-direction: column;
+    }
+
+    .flex-column-reverse--#{$breakpoint} {
+      flex-direction: column-reverse;
+    }
+  }
+}
+
+/*============================================================================*\
+   Wrap
+\*============================================================================*/
+@if $has-classes {
+  .flex-wrap {
+    flex-wrap: wrap;
+  }
+
+  .flex-nowrap {
+    flex-wrap: nowrap;
+  }
+
+  .flex-wrap-reverse {
+    flex-wrap: wrap-reverse;
+  }
+
+  @include for-each-breakpoints using ($breakpoint) {
+    .flex-wrap--#{$breakpoint} {
+      flex-wrap: wrap;
+    }
+
+    .flex-nowrap--#{$breakpoint} {
+      flex-wrap: nowrap;
+    }
+
+    .flex-wrap-reverse--#{$breakpoint} {
+      flex-wrap: wrap-reverse;
+    }
+  }
+}
+
+/*============================================================================*\
+   Grow / Shrink
+\*============================================================================*/
+$flex-grows: (0, 1) !default;
+$flex-shrinks: (0, 1) !default;
+$flexes: (0, auto, none) !default;
+
+@if $has-classes {
+  @each $value in $flex-grows {
```

Comment:
After some testing, the `clean-css` plugin used to minify the CSS is struggling to merge all the `@media` declarations in this file (see [this issue](https://github.com/jakubpawlowicz/clean-css/issues/687#issuecomment-269860944)). I suggest to loop once over each breakpoint to limit the number of `@media ...` in the compiled CSS. Something like this:

```sass
@each $value in $flex-grows {
  .flex-grow-#{$value} {
    flex-grow: $value;
  }
}

@each $value in $flex-shrinks {
  .flex-shrink-#{$value} {
    flex-shrink: $value;
  }
}

@each $value in $flexes {
  .flex-#{$value} {
    flex: $value;
  }
}

// Looping over each breakpoints once at the end
@include for-each-breakpoints using ($breakpoint) {
  @each $value in $flex-grows {
    .flex-grow-#{$value}--#{$breakpoint} {
      flex-grow: $value;
    }
  }

  @each $value in $flex-shrinks {
    .flex-shrink-#{$value}--#{$breakpoint} {
      flex-shrink: $value;
    }
  }

  @each $value in $flexes {
    .flex-#{$value}--#{$breakpoint} {
      flex: $value;
    }
  }
}
```